src/beamlime/applications/_nexus_helpers.py

```diff
@@ -2,10 +2,72 @@
 # Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
 import os
 from functools import partial
-from typing import Any, Callable, Literal, Optional, Union
+from typing import (
+    Any,
+    Callable,
+    Literal,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+)
 
 Path = Union[str, bytes, os.PathLike]
+DTypeDesc = Literal["int", "float", "string"]
+
+
+class DatasetRecipe(NamedTuple):
+    name: str
+    dtype_description: DTypeDesc
+
+
 FlatBufferIdentifier = Literal["ev44", "f144", "tdct"]
+GROUP_RECIPES: Mapping[FlatBufferIdentifier, Tuple[DatasetRecipe, ...]] = {
+    "f144": (DatasetRecipe("timestamp", "int"), DatasetRecipe("value", "float")),
+    "tdct": (DatasetRecipe("name", "string"), DatasetRecipe("timestamps", "int")),
+    "ev44": (
+        DatasetRecipe("reference_time", "int"),  # In nanoseconds
+        DatasetRecipe("reference_time_index", "int"),
+        DatasetRecipe("time_of_flight", "int"),  # In nanoseconds
+        DatasetRecipe("pixel_id", "int"),
+    ),
+}
+
+
+NestedObjectT = TypeVar("NestedObjectT", dict, list[dict])
+
+
+def nested_shallow_copy(obj: NestedObjectT, *indices) -> NestedObjectT:
+    """Shallow copy the nested dictionary and lists.
+
+    Examples
+    --------
+    >>> a = {'a': 1, 'b': {'b-a': 2}, 'c': {'c-a': 3}}
+    >>> b = nested_shallow_copy(a, 'b')
+    >>> b['b']['b-a'] = 3
+    >>> a['b']['b-a']  # The original dictionary is not changed
+    2
+    >>> b['b']['b-a']  # The copied dictionary is changed
+    3
+    >>> b['c']['c-a'] = 4
+    >>> a['c']['c-a']  # The original dictionary is also changed
+    4
+    """
+    from copy import copy
+
+    root_obj = obj.copy()
+    cur_parent = root_obj
+
+    try:
+        for idx in indices:
```

Comment:
So this is more like a path, not independent indices?

Reply:
Yes...! I'll rename it for clarity...